fe/fe-core/src/main/java/com/starrocks/connector/delta/DeltaLakeFileStats.java

```diff
@@ -0,0 +1,297 @@
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class DeltaLakeFileStats {
+    private static final DateTimeFormatter TIME_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS");
+    private static final DateTimeFormatter TIME_ZONE_FORMAT =
+            DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSXXX");
+
+    private final StructType schema;
+    private final List<String> nonPartitionPrimitiveColumns;
+    private long recordCount;
+    private long size;
+    private final Map<String, Object> minValues;
+    private final Map<String, Object> maxValues;
+    private final Map<String, Object> nullCounts;
+    private final Set<String> corruptedStats;
+    private boolean hasValidColumnMetrics;
+
+    public DeltaLakeFileStats(StructType schema,
+                              List<String> nonPartitionPrimitiveColumns,
+                              long recordCount,
+                              long size,
+                              Map<String, Object> minValues,
+                              Map<String, Object> maxValues,
+                              Map<String, Object> nullCounts) {
+        this.schema = schema;
+        this.nonPartitionPrimitiveColumns = nonPartitionPrimitiveColumns;
+        this.recordCount = recordCount;
+        this.size = size;
+        if (minValues == null || maxValues == null || nullCounts == null) {
+            this.minValues = null;
+            this.maxValues = null;
+            this.nullCounts = null;
+            this.corruptedStats = null;
+            this.hasValidColumnMetrics = false;
+        } else {
+            this.minValues = minValues;
+            this.maxValues = maxValues;
+            this.nullCounts = nullCounts;
+            this.corruptedStats = nonPartitionPrimitiveColumns.stream()
+                    .filter(col -> !minValues.containsKey(col) &&
+                            (!nullCounts.containsKey(col) || ((Double) nullCounts.get(col)).longValue() != recordCount))
+                    .collect(Collectors.toSet());
+            this.hasValidColumnMetrics = true;
+        }
+    }
+
+    public DeltaLakeFileStats(long recordCount) {
+        this.schema = null;
+        this.nonPartitionPrimitiveColumns = null;
+        this.recordCount = recordCount;
+        this.size = 0;
+        this.minValues = null;
+        this.maxValues = null;
+        this.nullCounts = null;
+        this.corruptedStats = null;
+        this.hasValidColumnMetrics = false;
+    }
+
+    public long getRecordCount() {
+        return recordCount;
+    }
+
+    public void fillColumnStats(ColumnStatistic.Builder builder, Column col) {
+        if (schema == null) {
+            return;
+        }
+
+        builder.setAverageRowSize(size * 1.0 / Math.max(recordCount, 1));
+        builder.setDistinctValuesCount(1);
```

Comment:
why ndv set to 1, this will affect the estimate of agg/join node

Reply:
Yes, there has riskey here.

Need also set builder.setType(ColumnStatistic.StatisticType.UNKNOWN);

similar as iceberge:

```
        Long ndv = colIdToNdv.get(fieldId);
        if (ndv != null) {
            builder.setDistinctValuesCount(Math.max(Math.min(ndv, icebergStats.getRecordCount()), 1));
            builder.setType(ColumnStatistic.StatisticType.ESTIMATE);
        } else {
            builder.setDistinctValuesCount(1);
            builder.setType(ColumnStatistic.StatisticType.UNKNOWN);
        }
```

Reply:
Fixed